source/grid/grid_tools.cc

```diff
@@ -1573,6 +1573,16 @@ namespace GridTools
       return false;
     }
 
+    template <int dim, typename Cell>
+    inline
+    double calculate_cell_radius(const Cell &cell)
+    {
+      double res = 0.;
+      for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
+        res = std::max(res, ( cell->vertex(v)).distance(cell->center()) );
```

Comment:
I think i have an example that demonstrates that it's indeed not the same thing. One can start from averaging of pairs of vertices on each diagonal. Then it's a matter of averaging those 4 (3D) or 2 (2D) points, located on each of the diagonals to get to the center. Then it becomes clear that the radius of the sphere to fully contain an element can be more than half of its diameter. See the sketch below (centers are determined via eyeballing!) .

![center_vertices](https://cloud.githubusercontent.com/assets/8023934/25269021/4268a264-267b-11e7-8e34-1f37638b20d0.png)
 
@vishalkenchan if this argument sounds reasonable to you as well, please switch back to the original code.